### PR TITLE
Add a For Loop mutator & test

### PR DESCRIPTION
**Pull request by jesperklippel**
This PR:

- [x] Adds new feature: For Loop Mutator
- [x] Covered by tests
- [x] Doc PR: https://github.com/infection/site/pull/27

This mutator serves the same purpose as the `ForEach_` mutator, except for For Loops.

### CODE REVIEW

File: src/Mutator/ZeroIteration/For_.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Mutator\ZeroIteration;
+
+use Infection\Mutator\Mutator;
+use PhpParser\Node;
+
+class For_ extends Mutator
```
Comment (theofidry) [inline src/Mutator/ZeroIteration/For_.php]: Do we need the trailing `_`?

Comment (jesperklippel) [reply src/Mutator/ZeroIteration/For_.php]: For is a reserved word by PHP, so it can't be used as a class name.

Comment (theofidry) [reply src/Mutator/ZeroIteration/For_.php]: `ForCondition`?

Comment (BackEndTea) [reply src/Mutator/ZeroIteration/For_.php]: `For_` is in line with all our other mutators

Comment (theofidry) [reply src/Mutator/ZeroIteration/For_.php]: Fair then :/

**Comment by borNfreee** [comment]
Thank you @jesperklippel

**Pull request merged**